pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java

```diff
@@ -54,13 +49,17 @@ public class TopicName implements ServiceUnitId {
 
     private final int partitionIndex;
 
-    private static final LoadingCache<String, TopicName> cache = CacheBuilder.newBuilder().maximumSize(100000)
-            .expireAfterAccess(30, TimeUnit.MINUTES).build(new CacheLoader<String, TopicName>() {
-                @Override
-                public TopicName load(String name) throws Exception {
-                    return new TopicName(name);
-                }
-            });
+    private static final ConcurrentHashMap<String, TopicName> cache = new ConcurrentHashMap<>();
```

Comment:
Before making this change, please compare performance to Caffeine Cache which is a successor for Guava Cache. We use Caffeine on many locations in Pulsar.

Reply:
Done, updated in the Motivation

Reply:
What's the difference to Caffeine?

Reply:
It's not a fair comparison unless Caffeine is simply used as a LRU cache without expiration.

Reply:
It is the same

[new_test.html.txt](https://github.com/user-attachments/files/16310771/new_test.html.txt)

![WechatIMG164_](https://github.com/user-attachments/assets/ae580645-a20a-42f0-8d4a-a08995c16252)

Reply:
Sorry, I forgot to disable `enableBrokerSideSubscriptionPatternEvaluation` when I tested with `Caffeine` before. I finished a new test and updated the result in the Motivation.

Summary:
- `with expireAfterXxx` cost more CPU resources
- `without expireAfterXxx` is close to `ConcurrentHashMap`